src/fem/discrete_space.hh

```diff
@@ -42,7 +43,7 @@ public:
   virtual unsigned int n_node_dofs(unsigned int nid) const = 0;
   
   /// Number of dofs associated to element (not shared by adjacent elements).
-  virtual unsigned int n_elem_dofs(const ElementAccessor<3> &cell) const = 0;
+  virtual unsigned int n_elem_dofs(const DHCellAccessor &cell) const = 0;
```

Comment:
Keep ElementAccessor in DiscreteSpace since it is constructed directly on Mesh and DH is then based on DiscreteSpace.